platform/saivpp/vpplib/SwitchStateBaseFdb.cpp

```diff
@@ -1082,3 +1082,352 @@ sai_status_t SwitchStateBase::vpp_delete_bvi_interface(
+    obj_type = objectTypeQuery(port_id);
+
+    if (obj_type != SAI_OBJECT_TYPE_PORT)
+    {
+        SWSS_LOG_NOTICE("SAI_BRIDGE_PORT_ATTR_PORT_ID=%s expected to be PORT but is: %s",
+                sai_serialize_object_id(port_id).c_str(),
+                sai_serialize_object_type(obj_type).c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    /* we need the VLAN ID attached based on the Port_ID */
+    sai_attribute_t attr;
+    attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
+
+    sai_status_t get_status = get(SAI_OBJECT_TYPE_PORT, port_id, 1, &attr);
+
+    if (get_status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("failed to get port vlan id from port %s",
+                sai_serialize_object_id(port_id).c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    uint32_t bd_id = attr.value.u16; /* bd_id is same as VLAN ID for .1Q bridge */
+
+    std::string ifname;
+
+    if (vpp_get_hwif_name(port_id, 0, ifname) == true) 
+    {
+        const char *hwif_name = ifname.c_str();
+        SWSS_LOG_NOTICE(" FDB_ENTRY delete on hwif_name: %s ", hwif_name);
+        auto ret = l2fib_add_del(hwif_name, fdb_entry.mac_address, bd_id, is_add, is_static);
+        SWSS_LOG_NOTICE(" FDB_Entry delete Success: %d", ret);
+
+    }
+    else
+    {
+        SWSS_LOG_ERROR("FDB entry Delete: Invalid ObjectID for the hwif on this bridge");
+
+        return SAI_STATUS_FAILURE;
+    }
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t SwitchStateBase::vpp_fdbentry_flush(
+                    _In_ sai_object_id_t switch_id,
+                    _In_ uint32_t attr_count,
+                    _In_ const sai_attribute_t *attr_list)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attribute;
+    sai_object_id_t br_port_id = 0;
+    sai_object_id_t port_id;
+    uint32_t bd_id = 0; 
+    uint8_t mode = 0;
+    bool is_static_entry = false;
+
+    for (uint32_t i = 0; i < attr_count; i++) 
+    {
+        attribute = attr_list[i];
+        switch (attribute.id) 
+        {
+            case SAI_FDB_FLUSH_ATTR_BRIDGE_PORT_ID:
+                {
+                    mode |=1;
+                    br_port_id = attribute.value.oid;
+                    sai_object_type_t obj_type = objectTypeQuery(br_port_id);
+
+                    if (obj_type != SAI_OBJECT_TYPE_BRIDGE_PORT)
+                    {
+                        SWSS_LOG_NOTICE("SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID=%s expected to be PORT but is: %s",
+                                sai_serialize_object_id(br_port_id).c_str(),
+                                sai_serialize_object_type(obj_type).c_str());
+
+                        return SAI_STATUS_FAILURE;
+                    }
+                }
+                break;
+            case SAI_FDB_FLUSH_ATTR_BV_ID:
+                {
+                    mode |= 2;
+                    bd_id = attribute.value.u16; 
+                }
+                break;
+
+            case SAI_FDB_FLUSH_ATTR_ENTRY_TYPE:
```

Comment:
We've considered the FLUSH of only DYNAMIC Entries from the SONiC side, hence adding this check to ensure only the DYNAMIC entries are flushed. Also, this is used for the FLUSH ALL case, where only TYPE ATTR should've been SET as part of FLUSH since port_ID/bd_ID is not passed.